# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/nsMIMEInfoImpl.h
# Commit: 1e4f8d9d96ce
# Full Hash: 1e4f8d9d96ce1ccb79c154416b119907f9479294
# Author: Mark Striemer <mstriemer@mozilla.com>
# Date: 2020-06-04 09:29:07
# Regressor Bug: 1632277
# File Overlap Count: 1
# Description:
#   Bug 1632277 - Part 1: Launch PDFs in app mode when default r=jaws,tkikuchi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73774
# ==============================================================================

diff -r 04d728d14947 -r 1e4f8d9d96ce uriloader/exthandler/nsMIMEInfoImpl.h
--- a/uriloader/exthandler/nsMIMEInfoImpl.h	Wed Jun 03 23:41:48 2020 +0000
+++ b/uriloader/exthandler/nsMIMEInfoImpl.h	Wed Jun 03 18:21:32 2020 +0000
@@ -66,6 +66,7 @@
       bool* aAlwaysAskBeforeHandling) override;
   NS_IMETHOD SetAlwaysAskBeforeHandling(bool aAlwaysAskBeforeHandling) override;
   NS_IMETHOD GetPossibleLocalHandlers(nsIArray** _retval) override;
+  NS_IMETHOD IsPdf(bool* isPdf);
 
   enum HandlerClass { eMIMEInfo, eProtocolInfo };
 
@@ -129,6 +130,8 @@
    */
   static nsresult LaunchWithIProcess(nsIFile* aApp, const nsCString& aArg);
   static nsresult LaunchWithIProcess(nsIFile* aApp, const nsString& aArg);
+  static nsresult LaunchWithIProcess(nsIFile* aApp, const int aArgc,
+                                     const char16_t** aArgv);
 
   /**
    * Given a file: nsIURI, return the associated nsIFile